source/common/network/dns_impl.h

```diff
@@ -53,6 +54,13 @@ class DnsResolverImpl : public DnsResolver {
     bool completed_ = false;
     // Was the query cancelled via cancel()?
     bool cancelled_ = false;
+    // If dns_lookup_family is "fallback", fallback to v4 address if v6
+    // resolution failed.
+    bool fallback_if_failed = false;
+    ares_channel channel_;
+    std::string dns_name_;
+
+    void getHostByName(int family);
```

Comment:
nit: put this right after void onAresHostCallback(int status, hostent* hostent);